### DISCUSSION TITLE
GPS Driver: supported GPS driver

### DISCUSSION THREAD

**Discussion by peterwillcn** [pendulum-project/ntpd-rs#1489] [Q&A] [answered]
Is it supported here（ https://www.ntp.org/documentation/drivers/ ）What about the GPS driver? Especially the Motorola Oncore GPS Receiver, a GPS time receiver with very high accuracy.

**Comment by JamesB192** [comment, marked as answer]
I do not work with the project, but they have been asked about this before. So synopsizing...

There are no plans to add ntp refclocks to ntpd-rs. Instead, the intent is to add a chrony type unix socket driver (not SHM); the one could run the [mlichvar/ntp-refclock](https://github.com/mlichvar/ntp-refclock) project, which supports many of the refclocks from ntpd. #882 #1138 #1361

Or did I hallucinate that?

**Comment by davidv1992** [comment]
Yes, we do intend to add a chrony type unix socket driver. Initial experiments on implementing this are currently ongoing and looking positive.

**Reply by peterwillcn** [reply]
Yes, Linux kernel drivers currently lack the aspect of Rust drivers. GPS drivers can be developed as a demonstration standard for the first introduction of the kernel, which can accelerate the integration of Rust into the Linux kernel.It is a good innovation attempt.

**Reply by davidv1992** [reply]
I think you misunderstand, these are not kernel drivers, but rather knowledge in the program itself to deal with certain types of hardware/software interfaces. Traditionally these have also been called drivers in the context of ntp.
